modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -74,13 +74,16 @@ const FrontendDataSet = ({
 	nestedItemsReferenceKey,
 	onActionDropdownItemClick,
 	onBulkActionItemClick,
+	onItemSelected,
```

Comment:
While I am be fine with pushing this without tests, I think unifying APIs should be a blocker. We should not add a new prop that does almost the same thing as an existing one. Instead:
- There should be only one callback, `onSelect`
- All views should use that one callback. That may imply removing current implementation in list view. This is used in fragment data set selection, where we can test any change.
- Implementations in current PR differ in the exposed data in callback. The old one implements a single selected item, new one a list of selected item. If there is a need for both features, we can expose both. To be used something like 
```js
onSelect({
    selectedItem: lastSelectedItem,
    selectedItems: allSelectedItems,
});
```
To simplify implementation, both can be optional.

Reply:
Hi @markocikos @dsanz 
If I'm not mistaken the agreement was to merge this to unblock Commerce feature, as the approach was correct, and then work on the needed changes. If there's any problem, please, can we talk on Monday to see what's needed to merge this?

Thanks!
cc. @gianmarcobrunialti